File: bin/emergency-stop-miniserver.php
#!/usr/bin/php
<?php
/* Read one configured Miniserver through the supported LoxBerry SDK only. */
require_once "loxberry_system.php";

$endpoint = $argv[1] ?? "";
if (!is_string($endpoint) || strlen($endpoint) > 512) { exit(2); }
$servers = LBSystem::get_miniservers();
foreach ($servers as $server) {
    if (!is_array($server)) { continue; }
    $host = $server['Ipaddress'] ?? '';
    $port = $server['Preferhttps'] ? ($server['Porthttps'] ?? 443) : ($server['Port'] ?? 80);
    $scheme = $server['Preferhttps'] ? 'https' : 'http';
    $candidate = $scheme . '://' . $host . (($scheme === 'https' && $port == 443) || ($scheme === 'http' && $port == 80) ? '' : ':' . $port);
    if ($candidate !== $endpoint) { continue; }
    $username = $server['Username'] ?? $server['User'] ?? '';
    $password = $server['Password'] ?? $server['Pass'] ?? '';
    if (!is_string($username) || !is_string($password) || $username === '' || $password === '') { exit(3); }
    echo json_encode(['username' => $username, 'password' => $password], JSON_UNESCAPED_SLASHES);
    exit(0);
}
exit(4);

File: bin/emergency-stop-miniserver.pl
#!/usr/bin/perl
use strict;
use warnings;

# Read one configured Miniserver through the supported LoxBerry SDK only.
use LoxBerry::System;
use JSON::PP;

my $endpoint = $ARGV[0] // '';
exit 2 if @ARGV != 1 || length($endpoint) > 512;

my %servers = LoxBerry::System::get_miniservers();
for my $number (sort { $a <=> $b } keys %servers) {
    my $server = $servers{$number};
    next if ref($server) ne 'HASH';
    my $host = $server->{IPAddress} // '';
    next if !defined($host) || $host eq '';
    $host = "[$host]" if $host =~ /:/ && $host !~ /\A\[.*\]\z/;
    my $scheme = $server->{Transport} // ($server->{PreferHttps} ? 'https' : 'http');
    next if $scheme ne 'http' && $scheme ne 'https';
    my $port = $scheme eq 'https' ? ($server->{PortHttps} // 443) : ($server->{Port} // 80);
    next if $port !~ /\A[0-9]{1,5}\z/ || $port < 1 || $port > 65535;
    my $default_port = $scheme eq 'https' ? 443 : 80;
    my $candidate = "$scheme://$host" . ($port == $default_port ? '' : ":$port");
    next if $candidate ne $endpoint;
    my $username = $server->{Admin} // '';
    my $password = $server->{Pass_RAW} // $server->{Pass} // '';
    exit 3 if ref($username) || ref($password) || $username eq '' || $password eq '';
    print encode_json({ username => $username, password => $password });
    exit 0;
}

exit 4;
